lib/rules/arrow-parens.js

```diff
@@ -80,29 +80,46 @@ module.exports = {
          * @returns {void}
          */
         function parens(node) {
-            const isAsync = node.async;
-            const firstTokenOfParam = sourceCode.getFirstToken(node, isAsync ? 1 : 0);
+            const firstToken = sourceCode.getFirstToken(node);
+            const arrowToken = sourceCode.getTokens(node).find(astUtils.isArrowToken);
+            const openingParenToken = astUtils.isOpeningParenToken(firstToken)
+                ? firstToken
+                : sourceCode.getTokensBetween(firstToken, arrowToken).find(astUtils.isOpeningParenToken);
```

Comment:
I think that `getFirstTokenBetween(nodeOrToken1, nodeOrToken2, skipOptions)` can be used here.

 [documentation link](https://github.com/eslint/eslint/blob/master/docs/developer-guide/working-with-rules.md#contextgetsourcecode)